### ISSUE TITLE
Flaky test: `AxHost_GetIFontDispFromFont_InvokeSimpleStyle_Roundtrips`

### ISSUE THREAD

**Issue by RussKie** [dotnet/winforms#2002]
* .NET Core Version: master @ 3750c43d5d9b5161be967e39fd848c89d6479ff5

**Problem description:**

`AxHost_GetIFontDispFromFont_InvokeSimpleStyle_Roundtrips` test fails locally from cli and VS:

```
  Message: 
    Assert.Equal() Failure
    Expected: 0
    Actual:   254
  Stack Trace: 
    AxHostTests.AxHost_GetIFontDispFromFont_InvokeSimpleStyle_Roundtrips() line 993
```

![image](https://user-images.githubusercontent.com/4403806/66012598-26ced180-e50b-11e9-81aa-e73d6b082e70.png)

`disp` doesn't appear to be cast-able to `Ole32.IFontDisp`.

**Expected behavior:**

The test pass

/cc: @hughbe

**Comment by weltkante** [comment]
As noted on #2003, is this test done on STA, being only annotated with a [Fact] attribute? I don't know if OLE ActiveX functions are working correctly outside STA ... considering that the OleInitialize native API initializes the thread to STA I'd expect all OLE functions to require STA ... in this case it might be solved together with #1999 by using the proper test annotations

**Comment by weltkante** [comment]
> disp doesn't appear to be cast-able to Ole32.IFontDisp

Wasn't able to reproduce this, is it failing consistently? Inconsistently failing casts really hint at the COM runtime being broken. So first thing to try once you have a machine where you can reproduce it is to use `[WinFormFact]` and place a call of `Application.OleRequired` at the begin of the test.

If this fixes the problem then [this PR](https://github.com/AArnott/Xunit.StaFact/pull/33) will be integrating the OleRequired call into WinFormFact so you don't have to do it manually anymore.
